### PR TITLE
GSG Human Orchestration Edits + Move cluster creation guides (Experiment 315 & 337)

### PR DESCRIPTION
**Pull request by Hafflgav**
According to [this issue](https://github.com/camunda/product-hub/issues/315) we want to move the cluster creation within the guides to a later point in time to avoid having so many stall clusters lying around. 

There fore I have moved the Cluster-Section and reframed it just a little bit. 
It would be great to get your feedback on this @christinaausley since you created these wonderful guides!

### CODE REVIEW

File: docs/guides/getting-started-orchestrate-microservices.md
Comment (christinaausley) [inline docs/guides/getting-started-orchestrate-microservices.md:42]: To confirm, is it correct that here we say you can essentially start an instance without having a cluster, but you cannot deploy a diagram without having a cluster?

**Review by christinaausley — CHANGES_REQUESTED** [review]
One comment for clarification on when exactly you need a cluster after developing a BPMN diagram, and one note on adding a link instead of bolding Console.

**Comment by akeller** [comment]
@christinaausley what are your thoughts about flow through the guides if the cluster creation step is moved? Do you think the flow is still relatively intuitive?

**Comment by christinaausley** [comment]
> @christinaausley what are your thoughts about flow through the guides if the cluster creation step is moved? Do you think the flow is still relatively intuitive?

Truthfully, initially my thought was it's a little awkward. To have a user create a diagram, leave that diagram to create a cluster, and then return to the diagram to actually deploy doesn't seem like the easiest user experience.

Ideally, I'd like to see a user be able to set up the entire environment required to actually develop and deploy their process (creating an account, setting up a cluster), and _then_ move into Modeler where they can create, run, and deploy their process without leaving the component.

Of course, I'm open to opposition and happy to discuss further.

**Comment by Camundaan** [comment]
> Truthfully, initially my thought was it's a little awkward. To have a user create a diagram, leave that diagram to create a cluster, and then return to the diagram to actually deploy doesn't seem like the easiest user experience.

That was also my thought initially @christinaausley and @akeller. Below is a description of why its a bit awkward right now.

**The awkwardness:**
Right now when a user has not created a cluster and clicks deploy in the modeler, there is a pop-up with a link to create a cluster. This will direct the user to the console in the same tab, leaving the modeler, hence they have to go find the modeler again after they created the cluster. Then go back into the project -> BPMN model -> hit deploy (again). The cluster might not be ready yet at that time either.

However, there are a couple of options to minimize the awkwardness:
1. Have the link create a new tab to the console to create the cluster, instead of using the same tab. This will leave the modeler open and make it easy to find for the user after they created the cluster. In the pop-up, we can communicate that creating a cluster might take a couple of minutes hence giving them the opportunity to go to some resources (Camunda academy, documentation, webinars) or have them "play around" in the modeler a little longer.

2. Same as above, but have a video play once they create the cluster that takes just as long as the average time it takes to create a cluster. This video will show/demonstrate the rest of the platform and what is possible. Example script "now that we are waiting for your cluster to be created, which takes an average of five minutes, we might as well take the time to show the rest of the platform to get an idea of the possibilities with Camunda 8. bla bla Operate, bla bla Optimize, bla bla tasklist. Ooh I see my 5 minutes are up, your cluster should be almost ready. Now that you have seen the platform we are excited to see what you orchestrate with Camunda 8." 

3. Have the creation of the cluster process in the modeler. This will probably take a lot of engineering resources to accomplish, hence not advisable, but it could be an option so they never have to leave the modeler. Maybe even have this combined with option 2.

I agree though, it is still not ideal, but having the users create the cluster first is costing us almost 100k per month, hence experimenting with alternatives can lead to cost savings in the end even when taking into account the "awkwardness" it might be worth the trouble.

Curious to hear your thoughts :)

**Comment by HanselIdes** [comment]
Hi @christinaausley and @Camundaan,

I've [recorded the desired flow](https://drive.google.com/file/d/1SrgCV3MtxMnbS5Ecm3LUtPwnwoYG3sK0/view?usp=sharing) and it's actually not as awkward as you might imagine. There's auto-redirect back to modeler

Timestamps (from an experienced user, with explanation):
- 0:07 - modeler loaded
- 0:09 - project loaded
- 0:17 - blank BPMN diagram created
- 0:27 - trivial human task process created
- 0:39 - navigate to console to create a new cluster
- 0:51 - new cluster requested
- 0:57 - modeler navigated to
  -  87 seconds of refreshing the page and waiting for my cluster...
- 2:24 - cluster ready
- 2:29 - diagram deployed
- 2:32 - process instance started
- 2:51 - operate, tasklist loaded
- 2:57 - task complete in tasklist
- 3:05 - process shows complete in Operate

**Comment by akeller** [comment]
@HanselIdes In your video, it looks like not only would we need to move the cluster creation steps, but we'd need to rewrite the documentation to include the link steps from Modeler to cluster creation and back again. Aare you looking to push this to production soon? You are right. This is less awkward and keeps the focus within web modeler, so I don't feel like I'm working with disparate components. The way this particular issue is written, this rewrite would be out of scope. 

In general, my goal with documentation is to always build habits with the user they can maintain throughout their experience with the product - not just during onboarding. If we always want users to model **then** create a cluster, let's move forward with this. If we are over-documenting an experience to save money, I'm going to continue to push back and ask that we look for other ways not to disrupt the experience, which I think Eric captured in his recording flow.

File: docs/guides/getting-started-orchestrate-microservices.md
Comment (akeller) [reply docs/guides/getting-started-orchestrate-microservices.md:42]: @christinaausley If you try to start an instance **or** deploy a diagram without a cluster you'll see a modal with a link to create a new cluster. Can you confirm this in your organization? It looks like this is in production, but I'm having trouble confirming (which prompted my message on this PR).

**Comment by HanselIdes** [comment]
Hi @akeller 

>In your video, it looks like not only would we need to move the cluster creation steps, but we'd need to rewrite the documentation to include the link steps from Modeler to cluster creation and back again. 
> In general, my goal with documentation is to always build habits with the user they can maintain throughout their experience with the product - not just during onboarding. If we always want users to model then create a cluster, let's move forward with this

It depends on how comprehensive/accurate we want the documentation to be.
I think we should take shortcuts during initial onboarding to get the user to their first strike ASAP, not to teach them how to use the product properly. However, I agree that documentation should teach good habits. 
I think we can strike a compromise by telling the user the proper thing to do (i.e. navigate to Console, then Cluster, then Create New Cluster), but then mentioning that for their first cluster, we've given them a shortcut that takes them directly to cluster creation and back.

> Are you looking to push this to production soon? You are right. This is less awkward and keeps the focus within web modeler, so I don't feel like I'm working with disparate components. The way this particular issue is written, this rewrite would be out of scope.

The feature (the shortcut link) already existed, so I'm simply making use of it.

**Comment by HanselIdes** [comment]
@akeller I've thought about this a bit more. I'm starting to wonder if we even _need_ to teach the user how to create a cluster throughout their getting started experience.
While they're on a trial plan, they can only create 1 cluster. Therefore, for the >99% of users that try out the product but _don't_ convert, teaching them this skill is moot. If their cluster gets automatically deleted due to inactivity, then they can use the shortcut again.
For customers that have already converted, I'm sure by that point they've gone through the Clusters page to navigate to the apps and seen the big blue "Create New Cluster" button. Plus, the shortcut takes them to the Clusters page, so they've at least seen it once.
image

> If we always want users to model then create a cluster, let's move forward with this.

Yes. This is the goal. Idle clusters just burn our cash, and the cluster creation step takes so little time compared to proper modeling that it's not a long-term issue. While the cluster is deploying, I'd like to add a video to give the user something to do while they wait.

**Comment by pepopowitz** [comment]
Hi friends!! I want to make sure you're aware of #1115 -- a PR that I plan to merge in the next week. 

It makes some fundamental changes to where our docs are sourced: 
1. 8.0 docs will now be sourced from `/versioned_docs/version-8.0` instead of `/docs`
2. Anything in `/docs` will be presented as the next/unreleased version

### Action required

It looks to me like this PR is intended to update the current 8.0 docs. Assuming that's true, one of these things need to happen: 

1. We need to merge this before I merge #1115. This is my favorite option if the PR is ready, because it means no extra work for any of us 😄 
2. After I merge #1115, we'll need to update this PR to make changes in the `/versioned_docs/version-8.0` folder instead of `/docs`. 

If this PR is intended for future 8.1 docs, we don't need to do anything! After #1115 merges, this PR _will_ be pointing at the unreleased 8.1 docs, and you can merge it at your convenience. 

---

I'll hold off on merging #1115 until at least August 8th, but probably not much after. If someone is able to get this one merged before then, great! If not, I'll assume that @Hafflgav, as the author of the PR, will update this PR as needed or ask for someone else to update it. 

Thanks! If you have questions or concerns, let me know.

**Comment by christinaausley** [comment]
Thanks for the update @pepopowitz! @Camundaan and I have an hour on the calendar tomorrow morning to polish this up and get it merged. If we want to iterate (particularly on the "Create a cluster" locations) we can always continue that in a separate PR.

**Comment by akeller** [comment]
Hi folks - I've taken over getting this PR cleaned up and merged. We have some local build errors and formatting issues I need to take care of. 

Once I get this working, I will make some small tweaks to additional areas of the docs and summarize what was done before merging.

File: docs/guides/assets/react-components/create-cluster.md
Comment (akeller) [reply docs/guides/assets/react-components/create-cluster.md:4]: Looks like there was no change to this in more recent commits, I've included this in the work I'm doing to finalize this PR.

File: docs/guides/assets/react-components/create-cluster.md
Comment (akeller) [reply docs/guides/assets/react-components/create-cluster.md:6]: I made a slight adjustment in my commits to finalize this PR.

**Comment by akeller** [comment]
* I cleaned up links for latest Docusaurus dependency updates.
  * This broke how links were being handled and was updated since this PR was opened. I wouldn't expect you to catch this if you aren't running the docs locally when making changes. @christinaausley and I can always be called into a PR to assist with build issues, particularly for infrequent docs authors.
  * Links can now be referenced from the section `/components/*` without `..` or `../..` or some other combination.
* Merged in `main` branch, again, because there were changes since this PR was opened, impacting local builds from being successful.
* I cleaned up for formatting issues identified by prettier.
* `create-cluster.md` was entirely removed from the docs.
  * If we remove pages, we need to redirect the traffic with rewrites/redirects via `.htaccess` and ensure they aren't linked from the product. **For these experiments, I would avoid deleting files.**
  * **I've restored this file and moved it to the "Next Steps" section**. Please keep in mind users outside the context of onboarding still need to have guides. This is why we compromised after #867 and created this section. 
  * I corrected the links to `create-cluster.md` to keep users in the context of Guides instead of routing them to Component content for Console. This is referenced as a React reusable component, but for maintaining docs this is cleaner IMO, unless we want to guide onboarding users in a different way than others, but it looks like that was out of scope of this PR.
    * Open a new PR if you want to modify cluster creation instructions specifically within the GSG.
    * Open a new PR if you want to modify the reusable component, which will update the instructions everywhere it's referenced. 
  * I didn't need to add a redirect because the sidebar sections (in this case) are not separated by directories (all files are in `/guides`. 

In general, keep your changes within the scope of the GSG. In documentation, we have different goals than onboarding for many of the other areas of the docs, which is why you can see me reverting changes for `create-cluster.md`. I'm happy to step the Onboarding team through the first and second iteration of the **Docs Vision** if understanding our goals in documentation would be helpful to understand better when and why we push back.

It adds some overhead, but we may want to approach this similar to what we've added for a docs review to the [PDP](https://confluence.camunda.com/pages/viewpage.action?spaceKey=HAN&title=Product+Development+Process#ProductDevelopmentProcess-DocumentationInvolvement) at the Define phase. This helps us in DX understand what is coming so we can make updates in parallel and coach the Onboarding team through docs and PR best practices.

**Comment by akeller** [comment]
I also added this to the Documentation board since Christina and I did more work than just a simple review. I'm not sure if you also want to add it to the Onboarding board, but I'll let the team decide that.

With these changes, I'll approve this PR and merge it. This does not mean it will go "live" immediately. I will coordinate with @pepopowitz to see if a docs release makes sense today and doesn't block him from anything he's working on. I'll send a message on Slack when this is live for transparency.

Thanks all!

**Review by akeller — APPROVED** [review]

**Pull request merged**
